src/components/delegation/Delegate.svelte

```diff
@@ -0,0 +1,213 @@
+  } from "../../utils/delegations.js";
+  import Swal from "sweetalert2";
+
+  let loading = false;
+  export let onDelegationChange;
+
+  let schema = yup.object().shape({
+    walletAddress: yup
+      .string()
+      .required("Wallet address is requred")
+      .label("Wallet address"),
+  });
+
+  let fields = {
+    walletAddress:
+      $veDelegation && $delegated > 0
+        ? `${$veDelegation.receiver.id?.substr(
+            0,
+            6
+          )}...${$veDelegation.receiver.id?.substr(
+            $veDelegation.receiver.id?.length - 6
+          )}`
+        : "",
+  };
+
+  const delegateVeOcean = async (values) => {
+    loading = true;
+    if (moment($oceanUnlockDate).diff(moment(), "days") < 7) {
+      errors.set({
+        walletAddress: "Current lock period must be grater than 1 week",
+      });
+      loading = false;
+      return;
+    }
+    try {
+      await delegate(
+        $userAddress,
+        values.walletAddress,
+        $oceanUnlockDate,
+        $networkSigner,
+        $veDelegation.createId
+      );
+    } catch (error) {
+      console.error(error);
+      Swal.fire("Error!", "Delegation failed.", "error");
+      loading = false;
+      return;
+    }
+    await onDelegationChange();
+    let newDelegated = await getDelegatedVeOcean($userAddress);
+    delegated.update(() => newDelegated);
+    let newVeOceansWithDelegations = await getUserVotingPowerWithDelegations(
+      $userAddress
+    );
+    veOceanWithDelegations.update(() => newVeOceansWithDelegations);
+    Swal.fire(
+      "Success!",
+      `You've delegated your veOCEAN allocation!`,
+      "success"
+    );
+    loading = false;
+  };
+
+  const cancelVeOceanDelegation = async () => {
+    loading = true;
+    try {
+      await cancelDelegation($veDelegation.tokenId, $networkSigner);
+    } catch (error) {
+      console.error(error);
+      Swal.fire("Error!", "Cancel delegation failed.", "error");
+      loading = false;
+      return;
+    }
+    await onDelegationChange();
+    let newDelegated = await getDelegatedVeOcean($userAddress);
+    delegated.update(() => newDelegated);
+    let newVeOceansWithDelegations = await getUserVotingPowerWithDelegations(
+      $userAddress
+    );
+    veOceanWithDelegations.update(() => newVeOceansWithDelegations);
+    Swal.fire(
+      "Success!",
+      `You've canceled your veOCEAN allocation!`,
+      "success"
+    );
+    loading = false;
+  };
+
+  const { form, errors, handleSubmit } = createForm({
+    initialValues: fields,
+    validationSchema: schema,
+    onSubmit: (values) => {
+      $veDelegation && delegated > 0
+        ? removeVeOceanDelegation($veDelegation.tokenId)
+        : delegateVeOcean(values);
+    },
+  });
+</script>
+
+<div class={`container`}>
+  <Card title="Delegate">
+    <p class="message">
+      You will delegate your entire veOCEAN allocation until your current lock
+      will end, and you can cancel at any time.
```

Comment:
"You will delegate your entire veOCEAN allocation until your current lock ends. You can cancel at any time."

Reply:
done